packages/orga/src/reader.ts

```diff
@@ -37,13 +33,13 @@ export const read = (text: string) => {
       const lp = linePosition(cursor.line)
       cursor = lp.end
     } else if (param === 'whitespaces') {
-      return eat(/^\s+/)
+      return eat(/^[ \t]+/)
     } else if (typeof param === 'number') {
       cursor = shift(start, param)
     } else {
-      const m = param.exec(substring({ start: cursor }))
+      const m = match(param, { start: cursor, end: eol() });
```

Comment:
When only a `start` is provided to `substring` it defaults to `eol()` -- `match` doesn't have an optional `end` here, so we're just providing it explictly. It should be exactly the same as before.

Reply:
Tbh it still feels weird that we provide a default `end` for `substring` (especially as it is `EOL`). I would've thought it would make more sense to either not provide _any_ default end, or make this be `EOF`.

In the case here, I think it is better to be more explicit anyway, otherwise we'd have to check the definition to find out where the `end` would be.